tests/jobserver/test_github.py

```diff
@@ -101,3 +109,63 @@ def test_get_repos_with_branches():
     assert len(output) == 1
     assert output[0]["name"] == "test-repo"
     assert output[0]["branches"][0] == "branch1"
+
+
+@responses.activate
+def test_githuborganizationoauth2_user_data_404():
+    expected_url = "https://api.github.com/orgs/opensafely/members/test-username"
+    responses.add(responses.GET, url=expected_url, status=404)
+
+    class DummyBackend(GithubOrganizationOAuth2):
+        def _user_data(*args, **kwargs):
+            return {"email": "test-email", "login": "test-username"}
+
+    with pytest.raises(AuthFailed, match="User doesn't belong to the organization"):
+        DummyBackend().user_data("access-token")
+
+
+@responses.activate
+def test_githuborganizationoauth2_user_data_302():
+    expected_url = "https://api.github.com/orgs/opensafely/members/test-username"
+    responses.add(responses.GET, url=expected_url, status=302)
+
+    class DummyBackend(GithubOrganizationOAuth2):
+        def _user_data(*args, **kwargs):
+            return {"email": "test-email", "login": "test-username"}
+
+    DummyBackend().user_data("access-token")
```

Comment:
Can you add a comment to the effect we just want to be sure it doesn't raise anything? Just for readability.